lenskit-core/src/main/java/org/grouplens/lenskit/data/event/Like.java

```diff
@@ -20,14 +20,120 @@
  */
 package org.grouplens.lenskit.data.event;
 
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+
+import java.io.Serializable;
+
 /**
  * Generic interface representing a unary interaction.  This provides a basic, generic representation
  * of likes, clicks, purchases, and other unary interactions.  Applications may want to implement
  * their own events to represent such interactions.
+ * <p>
+ * This class can be created with {@link LikeBuilder} or the {@link #create(long, long, long)} method.
+ * </p>
  *
  * @since 2.2
  * @see Events#like(long, long)
  * @see Events#like(long, long, long)
  */
-public interface Like extends Event {
+public final class Like implements Event, Serializable {
+    private static final long serialVersionUID = 1L;
+
+    private final long userId;
+    private final long itemId;
+    private final long timestamp;
+
+    Like(long uid, long iid, long ts) {
```

Comment:
I don't feel too strongly on this subject either. 
I think given that we have dependency injection, consistency on factories is out the window.  So Im not sure how much faith I would put on this. Where else do we use factories? (exhaustive list if possible, but if you really have the time it takes to make an exhaustive list I've got some concurrency bugs for you fix)
In the particular case of events I would air on the side of POJO. Its not worth asking people to think about these objects. (At that point, I guess supporting both and using whatever works best in context would be the way to go)